File: apps/mobile/ios/VoltClip/VoltClipClipboard.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(VoltClipClipboard, NSObject)

RCT_EXTERN_METHOD(getString:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getChangeCount:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: apps/mobile/ios/VoltClip/VoltClipClipboard.swift
import React
import UIKit

@objc(VoltClipClipboard)
class VoltClipClipboard: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool {
    true
  }

  @objc(getString:rejecter:)
  func getString(resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    resolve(UIPasteboard.general.string ?? "")
  }

  @objc(getChangeCount:rejecter:)
  func getChangeCount(resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    resolve(UIPasteboard.general.changeCount)
  }
}
